### PR TITLE
Source Asset -> Unexecutable Asset Adapter

### PR DESCRIPTION
**Pull request by schrockn**
## Summary & Motivation

Allow us to wrap an unexecutable source asset in a vanilla assets definition.

## How I Tested These Changes

BK

### CODE REVIEW

**Comment by schrockn** [comment]
Current dependencies on/for this PR:
* master
  * **PR #16617** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16617" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/16617?utm_source=stack-comment).

File: python_modules/dagster/dagster/_core/definitions/observable_asset.py [hunk 1]
```diff
@@ -50,3 +51,36 @@ def an_asset() -> None:
...
+        source_asset.auto_observe_interval_minutes is None,
+        "Observable source assets not supported yet: auto_observe_interval_minutes should be None",
+    )
+
+    kwargs = {
+        "key": source_asset.key,
+        "metadata": {
+            **source_asset.metadata,
+            **{SYSTEM_METADATA_KEY_ASSET_EXECUTION_TYPE: AssetExecutionType.UNEXECUTABLE.value},
+        },
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/definitions/observable_asset.py:71]: well need to ensure `is_source` get toggled on the `ExternalAssetNode` somehow, maybe other metadata or another update to the execution type enum

**Review by alangenfeld — COMMENTED** [review]
What exactly is the goal checking this in? Just check pointing that in process APIs work with the wrapper approach? 

We'll need something along the lines of https://github.com/dagster-io/dagster/pull/16637 to validate compatibility through the host process / UI experience. Does that just come later? Should we block this PR on covering that as well?

**Comment by schrockn** [comment]
Goal is to essentially enable writing that test case and making incremental progress on an implementation. I'm more than happy to move it to the test file itself or to park it.

**Review by alangenfeld — APPROVED** [review]
I think its fine to land the incremental progress so we can iterate. 

Can you update the summary accordingly. Just want to make it easy for someone in the future who bumps in to any loose ends left from this to be able to discern what was going on.

**Pull request merged**